### PR TITLE
[80] Setup RSpec & Guard

### PR DESCRIPTION
**Pull request by salcudeann**

### CODE REVIEW

**Review by drqCode — CHANGES_REQUESTED** [review]
Please write a dummy test

**Comment by drqCode** [comment]
image
 Let's avoid using deprecations

File: Gemfile [hunk 1]
```diff
@@ -42,6 +43,8 @@ group :development do
   gem 'listen', '~> 3.3'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
+  gem 'guard', '~> 2.18.0'
+  gem 'rexml'
```
Comment (alex-damian-negru) [inline Gemfile:47]: Is this gem needed?

File: Guardfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+## Note: if you are using the `directories` clause above and you are not
+## watching the project directory ('.'), then you will want to move
+## the Guardfile to a watched dir and symlink it back, e.g.
+#
+#  $ mkdir config
+#  $ mv Guardfile config/
+#  $ ln -s config/Guardfile .
+#
+# and, you'll have to watch "config/Guardfile" instead of "Guardfile"
```
Comment (alex-damian-negru) [inline Guardfile:16]: Guard is useless at this point, it should be configured to run RSpec on file changes.

Have a look over https://github.com/guard/guard-rspec.

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (salcudeann) [reply Gemfile:47]: Yes, without `rexml` i get some errors.

Comment (alex-damian-negru) [reply Gemfile]: As we saw, `rexml` seems to be needed only if we run the tests through `rake`. `bundle exec guard` seems to work fine without them.

We'll therefore remove this gem & add it later if errors arise.

**Comment by alex-damian-negru** [comment]
Please rebase this on `main`.

**Review by alex-damian-negru — APPROVED** [review]

**Comment by salcudeann** [comment]
[API: Setup testing](https://trello.com/c/sFsCauOp/80-api-setup-testing)

**Pull request merged**